src/librustc/Cargo.toml

```diff
@@ -27,6 +27,7 @@ syntax = { path = "../libsyntax" }
 syntax_pos = { path = "../libsyntax_pos" }
 backtrace = "0.3.3"
 byteorder = { version = "1.1", features = ["i128"]}
+getopts = "0.2"
```

Comment:
How come this is necessary? I believe this means that we're building and shipping two copies of getopts, right? Was the one copy previously causing problems?

Reply:
It is more accurate and probably I good idea with https://github.com/rust-lang/rust/pull/49119, since otherwise librustc will depend on the bootstrap compiler's `getopts`, which makes updating this dependency tricky.

Reply:
We should also make sure compiler crates can depend on crates.io crates which in turn depend on `getopts`. @nikomatsakis ran into problems adding such a dependency.

Reply:
Isn't this what `rustc_cratesio_shim` fixes?

Reply:
In general no compiler crate should depend on `getopts` because nothing should be doing argument parsing other than rustc itself. @eddyb that's more for the dylib situation rather than depending on things at all, this is a little different because `getopts` is currently inherited through libtest without an explicit dependency

Reply:
I think this can be fixed by having `rustc.exe` remove `--extern=getopts` arguments so crates.io crates are also forced to use the `getopts` in the sysroot.